src/pwm.rs

```diff
@@ -138,117 +138,159 @@ use crate::pac::{TIM2, TIM3, TIM4};
 
 use crate::afio::MAPR;
 use crate::bb;
-use crate::gpio::gpioa::{PA0, PA1, PA2, PA3, PA6, PA7};
-use crate::gpio::gpiob::{PB0, PB1, PB6, PB7, PB8, PB9};
+use crate::gpio::gpioa::{PA0, PA1, PA2, PA3, PA6, PA7, PA15};
+use crate::gpio::gpiob::{PB0, PB1, PB3, PB4, PB5, PB6, PB7, PB8, PB9, PB10, PB11};
+use crate::gpio::gpioc::{PC6, PC7, PC8, PC9};
 use crate::gpio::{Alternate, PushPull};
 use crate::time::Hertz;
 use crate::timer::Timer;
 
+
+mod sealed {
+    pub trait Remap {
+        type Periph;
+        const REMAP: u8;
+    }
+    pub trait Ch1<REMAP> {
+        const ENABLED: bool;
+    }
+    pub trait Ch2<REMAP> {
+        const ENABLED: bool;
+    }
+    pub trait Ch3<REMAP> {
+        const ENABLED: bool;
+    }
+    pub trait Ch4<REMAP> {
+        const ENABLED: bool;
+    }
+}
+use sealed::{Remap, Ch1, Ch2, Ch3, Ch4};
+
 pub trait Pins<TIM> {
-    const REMAP: u8;
-    const C1: bool;
-    const C2: bool;
-    const C3: bool;
-    const C4: bool;
+    fn ch1_state() -> bool;
+    fn ch2_state() -> bool;
+    fn ch3_state() -> bool;
+    fn ch4_state() -> bool;
     type Channels;
 }
-
-impl Pins<TIM2>
-    for (
-        PA0<Alternate<PushPull>>,
-        PA1<Alternate<PushPull>>,
-        PA2<Alternate<PushPull>>,
-        PA3<Alternate<PushPull>>,
-    )
-{
-    const REMAP: u8 = 0b00;
-    const C1: bool = true;
-    const C2: bool = true;
-    const C3: bool = true;
-    const C4: bool = true;
-    type Channels = (Pwm<TIM2, C1>, Pwm<TIM2, C2>, Pwm<TIM2, C3>, Pwm<TIM2, C4>);
+impl<TIM, REMAP, CH1, CH2, CH3, CH4> Pins<REMAP> for (CH1, CH2, CH3, CH4)
+where
+    REMAP: Remap<Periph = TIM>,
+    CH1: Ch1<REMAP>,
+    CH2: Ch2<REMAP>,
+    CH3: Ch3<REMAP>,
+    CH4: Ch4<REMAP> {
+    type Channels = (Pwm<TIM, C1>, Pwm<TIM, C2>, Pwm<TIM, C3>, Pwm<TIM, C4>);
```

Comment:
Unless I am missing something, C1-C4 channel specifiers become useless after your changes, since channel enablement is done based on chX_state()

Reply:
`NoPin` has `false` value.

Reply:
Correct. But then we don't need C1-C4 phantoms.